l10n_ch_dta/wizard/create_dta.py

```diff
@@ -545,35 +546,36 @@ class DTAFileGenerator(models.TransientModel):
     @api.model
     def _initialize_elec_context(self, data):
         elec_context = {}
-        payment_obj = self.env['payment.order']
+        payment_obj = self.env['payment.register']
         payment = payment_obj.browse(data['id'])
         elec_context['uid'] = str(self.env.uid)
         elec_context['creation_date'] = time.strftime('%y%m%d')
-        if not payment.mode:
+        if not payment.mode.journal_id.type:
             raise except_orm(
                 _('Error'),
                 _('No payment mode')
             )
-        bank = payment.mode.bank_id
+        res_bank = payment.mode.bank_id
+        bank = res_bank.bank_id
         if not bank:
```

Comment:
This test should be:

`if not res_bank:`

Otherwise the secont test on `bank` is useless